sway/desktop/transaction.c

```diff
@@ -104,8 +108,16 @@ static void copy_workspace_state(struct sway_workspace *ws,
 	state->layout = ws->layout;
 
 	state->output = ws->output;
-	state->floating = create_list();
-	state->tiling = create_list();
+	if (state->floating) {
+		state->floating->length = 0;
```

Comment:
Why are these changes now necessary? Is it just a bugfix for leaked memory?

If so, please split these into a separate commit.

Reply:
Ah, my bad, this is required for the new `transaction_add_node` logic.